plugins/techdocs-backend/src/service/router.ts

```diff
@@ -127,6 +132,15 @@ export async function createRouter({
 
     const entity: Entity = await catalogRes.json();
 
+    if (!entity.metadata.uid) {
+      res.status(404).json({ error: 'Entity metadata UID missing' });
```

Comment:
just noting that we now have standardised error handling in the backend, so just throwing a `NotFoundError` from `@backstage/errors` will give the caller a rich, standard feedback mechanism to lean on

Reply:
I'll start using them in the frontend code

Reply:
PR updated using methods from @backstage/errors